dbflow/src/main/java/com/raizlabs/android/dbflow/structure/ModelAdapter.java

```diff
@@ -63,12 +57,8 @@ public DatabaseStatement getInsertStatement(DatabaseWrapper databaseWrapper) {
      * @return The precompiled full statement for this table model adapter
      */
     public DatabaseStatement getCompiledStatement() {
-        if (compiledStatement == null) {
-            compiledStatement = getCompiledStatement(
-                    FlowManager.getDatabaseForTable(getModelClass()).getWritableDatabase());
-        }
-
-        return compiledStatement;
+        return getCompiledStatement(
```

Comment:
Now I get it. :) Thanks for the clarification. Yes, that absolutely makes
sense. Will try to implement something like that tomorrow.

Andrew Grosner <notifications@github.com> schrieb am Di., 3. Mai 2016,
20:58:

> In
> dbflow/src/main/java/com/raizlabs/android/dbflow/structure/ModelAdapter.java
> <https://github.com/Raizlabs/DBFlow/pull/757#discussion_r61936260>:
>
> > @@ -63,12 +57,8 @@ public DatabaseStatement getInsertStatement(DatabaseWrapper databaseWrapper) {
> >       * @return The precompiled full statement for this table model adapter
> >       */
> >      public DatabaseStatement getCompiledStatement() {
> > -        if (compiledStatement == null) {
> > -            compiledStatement = getCompiledStatement(
> > -                    FlowManager.getDatabaseForTable(getModelClass()).getWritableDatabase());
> > -        }
> > -
> > -        return compiledStatement;
> > +        return getCompiledStatement(
>
> What I meant is we should delay closing it until it sits idle on the main
> thread. But while thats happening we should not regenerate the
> SQLiteStatement every method call here. This will improve performance
> such that we can rapidly save stuff without reconstructing it everytime.
> Itll close it after a bulk op occurs and recreate it next time needed. I
> think that makes sense, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Raizlabs/DBFlow/pull/757/files/6e10e4f2aaa682fd69dbcdfc6168680edf43b002#r61936260>
>